### DISCUSSION TITLE
Commands through a Window not working

### DISCUSSION THREAD

**Discussion by VinRubio** [spyoungtech/ahk#334] [Q&A]
I have the bellow code, where when I try to send a key or a click using the window not work, if I change to use ahk instead of win, it works perfectly, but now using a window

    win = ahk.find_window(title="Calculator")
    
    time.sleep(0.1)

    win.send("5")
    win.click(x=159, y=417)

I used other methods to know if the right window is being selected and all of them shows that is right window, the calculator in this case.

**Comment by spyoungtech** [comment]
If you are using AutoHotkey v2, chances are you have to specify the control you want to send to. I'm guessing that's the issue you're running into. This is described in the README section describing [differences between AHK v1 and v2](https://github.com/spyoungtech/ahk?tab=readme-ov-file#differences-when-using-autohotkey-v1-vs-autohotkey-v2).

In AutoHotkey v1, the default is to send to the topmost control, which usually does the correct thing.

For example:


In AutoHotkey v1 this works:

```py
ahk = AHK(version='v1')
notepad = ahk.find_window(title='Untitled - Notepad')
notepad.send('hello')
```

But in AutoHotkey version 2, the above does not work and you need to specify the control:

```python
ahk = AHK(version='v2')
notepad = ahk.find_window(title='Untitled - Notepad')
notepad.send('hello', control='Edit1')
```

You can use the `.list_controls` method to find controls of the window. See if either adding the `control` parameter to the `send` method or using AutoHotkey v1 instead works.

**Reply by VinRubio** [reply]
> If you are using AutoHotkey v2, chances are you have to specify the control you want to send to. I'm guessing that's the issue you're running into. This is described in the README section describing [differences between AHK v1 and v2](https://github.com/spyoungtech/ahk?tab=readme-ov-file#differences-when-using-autohotkey-v1-vs-autohotkey-v2).
> 
> In AutoHotkey v1, the default is to send to the topmost control, which usually does the correct thing.
> 
> For example:
> 
> In AutoHotkey v1 this works:
> 
> ```python
> ahk = AHK(version='v1')
> notepad = ahk.find_window(title='Untitled - Notepad')
> notepad.send('hello')
> ```
> 
> But in AutoHotkey version 2, the above does not work and you need to specify the control:
> 
> ```python
> ahk = AHK(version='v2')
> notepad = ahk.find_window(title='Untitled - Notepad')
> notepad.send('hello', control='Edit1')
> ```
> 
> You can use the `.list_controls` method to find controls of the window. See if either adding the `control` parameter to the `send` method or using AutoHotkey v1 instead works.

Thank you for the help, I will test this, I don't know which version I'm using lol, I only used the command to install the binaries following the docs and I don't chosen any version when I instatiate the AHK.
